Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -215,6 +207,7 @@ void TimeAndSalesTableView::on_vertical_slider_value_changed(
   }
   m_model->set_row_visible(m_table->rowAt(
     m_table->visibleRegion().boundingRect().bottom()));
+  m_header->viewport()->update();
```

Comment:
Oh much better and it makes a lot more sense, it directly ties the visual update to a change in the data model.